packages/vitest/src/node/pool.ts

```diff
@@ -74,32 +39,54 @@ export function createWorkerPool(ctx: Vitest): WorkerPool {
     options.concurrentTasksPerWorker = 1
   }
 
+  if (!ctx.config.threads) {
+    options.concurrentTasksPerWorker = 1
+    options.maxThreads = 1
+    options.minThreads = 1
+  }
+
+  if (ctx.config.coverage)
+    process.env.NODE_V8_COVERAGE ||= ctx.config.coverage.tempDirectory
+
+  options.env = {
+    TEST: 'true',
+    VITEST: 'true',
+    NODE_ENV: ctx.config.mode || 'test',
+    VITEST_MODE: ctx.config.watch ? 'WATCH' : 'RUN',
+    ...process.env,
+    ...ctx.config.env,
+  }
+
   const pool = new Tinypool(options)
 
-  const runWithFiles = (name: string): RunWithFiles => {
-    return async (files, invalidates) => {
-      let id = 0
-      const config = ctx.getSerializableConfig()
-      await Promise.all(files.map(async (file) => {
-        const { workerPort, port } = createChannel(ctx)
-
-        const data: WorkerContext = {
-          port: workerPort,
-          config,
-          files: [file],
-          invalidates,
-          id: ++id,
-        }
+  const runTests: RunWithFiles = async (files, invalidates) => {
+    let id = 0
+    const config = ctx.getSerializableConfig()
+
+    const runFiles = async (files: string[]): Promise<void> => {
+      const { workerPort, port } = createChannel(ctx)
+
+      const data: WorkerContext = {
+        port: workerPort,
+        config,
+        files,
+        invalidates,
+        id: ++id,
+      }
 
-        await pool.run(data, { transferList: [workerPort], name })
-        port.close()
-        workerPort.close()
-      }))
+      await pool.run(data, { transferList: [workerPort], name: 'run' })
+      port.close()
+      workerPort.close()
     }
+
+    if (!ctx.config.isolate && !ctx.config.threads)
+      await runFiles(files) // single thread with no isolation
```

Comment:
I think `--no-threads` should always run all files in a single worker, otherwise there is no point in `--no-threads`. It's _literally_ the same as running `--threads` - no alternative. And the name becomes misleading.

I think `--isolate` should just not affect `--no-threads`. I think in the future we might use `vm.runInContext` for isolation in `--no-threads`.